cli/reactjs_jsx_ppx.ml

```diff
@@ -2689,6 +2697,7 @@ let getMapper ~config =
   in
   let signature mapper items =
     let oldConfig = saveConfig () in
+    let hasReactComponent = ref false in
```

Comment:
If you put this in config, then you get it restored for free after sub-modules are done.

Reply:
Yes, it will work. I thought that too, but I little hesitated because the config means the config of jsx. It seems little unmatched with `hasReactComponent`, doesn't it?

Reply:
hmm.. There is `nestedModules` inside the config already. I think `hasReactComponent` could be there too. I'll fix it.

Reply:
Yes `nestedModules` is in there for exactly this reason: avoid passing ad-hoc state when it can be all together.
If you think the type should have another better name, go ahead renaming it.

Reply:
I fixed it by adding `hasReactComponent` into the config. I realized why I felt a little weird about it.
It is because the config keeps the values until it hits another config update, even though it is restored when it goes to the upper level, but it doesn't set default when it goes to the deeper level. Therefore config.hasReactComponent should be set default(false) in the structure and signature mapper.
I think `hasReactComponent` doesn't fit in the config.

Reply:
This is not very important. You could put in inside or leave it outside.

Reply:
`module_binding` is only for implementations.
I've quickly edited so it supports interfaces as well.

Reply:
Ah! Good catch. Thanks!